components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts

```diff
@@ -41,33 +41,33 @@ export const rewardsPanelReducer = (state: RewardsExtension.State | undefined, a
       state.walletCreated = false
       state.walletCorrupted = false
       break
-    case types.ON_WALLET_CREATED:
+    case types.ON_WALLET_INITIALIZED: {
+      const result = payload.result
       state = { ...state }
-      state.walletCreated = true
-      state.walletCreateFailed = false
-      state.walletCreating = false
-      state.walletCorrupted = false
-      chrome.braveRewards.saveAdsSetting('adsEnabled', 'true')
-      chrome.storage.local.get(['is_dismissed'], function (result) {
-        if (result && result['is_dismissed'] === 'false') {
-          chrome.browserAction.setBadgeText({
-            text: ''
-          })
-          chrome.storage.local.remove(['is_dismissed'])
-        }
-      })
-      break
-    case types.ON_WALLET_CREATE_FAILED:
-      state = { ...state }
-      state.walletCreateFailed = true
-      state.walletCreating = false
-      state.walletCreated = false
-      state.walletCorrupted = false
-      break
-    case types.ON_WALLET_CORRUPTED:
-      state = { ...state }
-      state.walletCorrupted = true
+      if (result === 12) {  // ledger::Result::WALLET_CREATED
```

Comment:
Should these results be in some kind of enumeration?

Reply:
The problem I was seeing is that this result comes from an enum in Ledger. Currently we would have to keep track of making sure the Ledger enum and an enum here would stay in sync.

Reply:
we already do this and you have definition in `rewards.d.ts` -> `Rewards.Result`

Reply:
Nice! Never saw that

Reply:
Sadly, per @NejcZdovc there is currently a problem using ts enum in reducer

Reply:
The enum declaration needs to be exported as `const` in order to work properly. Done.